skycoin/skywire
# Versioned visor configs.

Fixes #362 Fixes #365 

 Changes:	
* Moved visor conifg-related parts to a new module /pkg/visor/visorconfig.
* Introduced visorconfig.Common that contains shared fields across all config versions.
* Improved config-parsing logic.

To do:
- [x] Visor shutdown order should be inverted.
- [x] Implement default `testenv` config generation.
- [x] Implement config upgrading from `v0.0.0`.
- [x] Fix CLI: https://discordapp.com/channels/466450852098605059/466450852098605061/711893952856457237
    ```
    $ ./skywire-cli visor add-tp 0219bec686b86f828bd1995d6c5c8587aa6d55a3f1ae88951eda0890ca456e4d08 --type dmsg
    [2020-05-18T13:46:12+03:00] FATAL [skywire-cli]: Failed to establish dmsg transport error="rpc client timeout"
    ```
- [x] Add `pk` back into visor config.
- [x] Make all package loggers respect master logger settings.

How to test this PR:
* `make check`
* `skywire-cli visor gen-config` (test with different combinations of flags)
* `skywire-visor` with an older config version (e.g. from `master` branch)
